### DISCUSSION TITLE
Cors issue when trying to access a different port via URL

### DISCUSSION THREAD

**Discussion by litcodeur** [coder/code-server#6080] [Q&A]
Hello, I'm running into cors issue as follows, when trying to access my API running on port 5100 from a react app running on port 3002.
```
Access to fetch at 'https://5100--myurl.com/route' from origin 'https://3002--myurl.com' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: Redirect is not allowed for a preflight request.
```

I'm using native Javascript fetch to make the request and tried using the `{ "credentials": "include" }` header but that doesn't seem to help either.

Is there a way to request resource from the api using the react app using just url? Or would the best path forward would be to expose the port directly and use the ip address of the code-server to run the request on?

Let me know if more context is required on my current setup if it helps.

Thanks!

**Comment by code-asher** [comment]
It looks like this might be coder rather than code-server?  Just by going off the `{port}--` syntax, but apologies if I have it wrong.

The `Redirect is not allowed for a preflight request` is interesting, I suppose it is redirecting to the login page?  Maybe the API needs the `Access-Control-Allow-Credentials` header?

https://stackoverflow.com/a/24689738

**Reply by litcodeur** [reply]
Thanks @code-asher, I think it might be a coder issue too but I didn't know where it'd fit better so I posted it here initially. I have the header setup in my api server to set the `Access-Control-Allow-Credentials` to "*". But it's the preflight request to the origin that is failing because I believe that the coder instance is trying to verify that only authenticated users can visit the domain exposed and since pre-flight requests don't include any cookies it is failing to redirect it correctly.

I'm moving this over to coder discussion thread as I think it'd be more appropriate there.

**Reply by litcodeur** [reply]
I also noticed that it is an open issue on the coder repository [#5706](https://github.com/coder/coder/issues/5706), so that answers my question. Thanks!

**Reply by code-asher** [reply]
Oh nice find!!

Ahhhh that makes sense.
